driver-core/src/test/unit/com/mongodb/internal/connection/AuthorizationHeaderTest.java

```diff
@@ -66,6 +66,26 @@ public void testGetRegion() throws SaslException {
         assertEquals(expected, actual);
     }
 
+    @Test(expected = SaslException.class)
+    public void shouldThrowErrorOnEmptyMiddleHostLabels() throws SaslException {
+        AuthorizationHeader.getRegion("abc..def");
+    }
+
+    @Test(expected = SaslException.class)
+    public void shouldThrowErrorOnEmptyHostLabelAtStart() throws SaslException {
+        AuthorizationHeader.getRegion(".abc.def");
+    }
+
+    @Test(expected = SaslException.class)
+    public void shouldThrowErrorOnEmptyHostLabelAtEnd() throws SaslException {
+        AuthorizationHeader.getRegion("abc.def.");
+    }
+
+    @Test(expected = SaslException.class)
+    public void shouldThrowErrorOnEmptyHost() throws SaslException {
+        AuthorizationHeader.getRegion("");
+    }
+
```

Comment:
These tests just make sure that empty values throw exceptions per the [spec](https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#id11).